src/client/user/UserHero.js

```diff
@@ -29,64 +40,91 @@ class UserMenuWrapper extends React.Component {
   }
 }
 
-const UserHero = ({
-  authenticated,
-  user,
-  username,
-  isSameUser,
-  coverImage,
-  hasCover,
-  isFollowing,
-  onTransferClick,
-}) => (
-  <div>
-    <Switch>
-      <Route
-        path="/@:name"
-        render={() => (
-          <div>
-            {user.fetching ? (
-              <UserHeaderLoading />
-            ) : (
-              <UserHeader
-                username={username}
-                handle={user.name}
-                userReputation={user.reputation}
-                vestingShares={parseFloat(user.vesting_shares)}
-                isSameUser={isSameUser}
-                coverImage={coverImage}
-                hasCover={hasCover}
-                isFollowing={isFollowing}
-                onTransferClick={onTransferClick}
-              />
-            )}
-            <UserMenuWrapper followers={user.follower_count} following={user.following_count} />
-          </div>
-        )}
-      />
-      <Route render={() => (authenticated ? <Hero /> : <div />)} />
-    </Switch>
-  </div>
-);
+// eslint-disable-next-line react/no-multi-comp
+@connect(
+  state => ({
+    time: getGlobalPropertiesTime(state),
```

Comment:
Why you are requesting the blockchain to know the time? It feel like adding a lot of complexity for nothing, cuz we have the time without it.

Reply:
Last action must be base on blockchain time otherwise it won't work. For instance I got a 2 hours difference between my computer and the blockchain so if I use `new Date()` only the activity is already 2 hours old and the activity indicator never shows